ph5/clients/ph5availability.py

```diff
@@ -129,23 +135,40 @@ def get_array_order_id(self, array_name):
         arrayorder = self.ph5.Array_t[array_name]['order']
         return arrayorder, arraybyid
 
-    def get_time_das_t(self, das, start, end, component=None):
+    def get_time_das_t(self, das, start, end,
+                       component=None, sample_rate=None):
         s = 0 if start is None else start
         e = 32509613590 if end is None else end
-        self.ph5.read_das_t(das, s, e, reread=False)
-        if das not in self.ph5.Das_t:
-            if (start is not None) and (end is not None):
-                LOGGER.warning("No Das table found for %s in range %s - %s"
-                               % (das, start, end))
-            return -1
-        if component is None:
-            Das_t = deepcopy(self.ph5.Das_t[das]['rows'])
+        if sample_rate is not None:
+            Das_t = self.ph5.query_das_t(
+                das,
+                chan=component,
+                start_epoch=s,
+                stop_epoch=e,
+                sample_rate=sample_rate)
+            if not Das_t:
+                if (start is not None) and (end is not None):
+                    LOGGER.warning(
+                        "No Das table found for %s in range %s - %s for "
+                        "component: %s, samplerate:%s"
+                        % (das, start, end, component, sample_rate))
+                return -1
         else:
-            Das_t = ph5api.filter_das_t(self.ph5.Das_t[das]['rows'], component)
+            self.ph5.read_das_t(das, s, e, reread=False)
+            if das not in self.ph5.Das_t:
+                if (start is not None) and (end is not None):
+                    LOGGER.warning("No Das table found for %s in range %s - %s"
+                                   % (das, start, end))
+                return -1
+            Das_t = deepcopy(self.ph5.Das_t[das]['rows'])
```

Comment:
Is this `deepcopy` call really necessary? This seems expensive.

Reply:
I don't think deepcopy is needed here.

Reply:
I will remove that. I was afraid the normal copy doesn't work after forget_das_t but actually it still work after removing deepcopy.

Reply:
@damhuonglan Are you sure a copy is necessary at all? I think it should be avoided completely if possible.

Reply:
`Das_t = self.ph5.Das_t[das]['rows']` should be okay. Lists are passed by reference so no copying is being done. But I think you are using self.ph5.forget_das_t(das) a bit early. You move that to the end of the method for safety.